minimum_match accepts not only int but string

I was trying to run a `Terms` query with `minimum_match` and passing a percentage but was getting "wrong" results. Inspected the method and noticed that its only accepting `int`.

I can't find anything on the ElasticSearch docu about this (only finding `minimum_should_match`) but I ran manually a query against ES and confirmed that you can send "minimum_match" as a string (percentage).

Before doing a PR, I'd like to confirm this.

I'm using `Elastica 2.3.*` - Still being `int` in `3.*`

Hi @ruflin,

I tried contacting you outside GitHub but no luck (Twitter).

I'm experiencing this issue when running the tests:

```
☁  Elastica [fix-minimum-match-type-hinting|MERGING] ⚡ make run RUN="make phpunit"
docker run -v /path/to/repo/Elastica:/elastica ruflin/elastica make phpunit
Waiting on elasticsearch(elasticsearch:9200) to start...............................
Elasticsearch is not running or is not healthy.
Address: elasticsearch:9200

make: *** [run] Error 1
```

Any idea?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@ All notable changes to this project will be documented in this file based on the
 
 ### Bugfixes
 - Set HTTP headers on each request preventing server error if persistent connection is enabled and compression enabled and later disabled for the same connection.
+- Removed `int` type hinting in `setMinimumMatch` (`Terms` Query): it should also allow `string`. [#1151](https://github.com/ruflin/Elastica/pull/1151)  
 
 ### Added
 - Elastica\QueryBuilder\DSL\Query::geo_distance

--- a/lib/Elastica/Query/Terms.php
+++ b/lib/Elastica/Query/Terms.php
@@ -7,6 +7,7 @@ use Elastica\Exception\InvalidException;
  * Terms query.
  *
  * @author Nicolas Ruflin <spam@ruflin.com>
+ * @author Roberto Nygaard <roberto@nygaard.es>
  *
  * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html
  */
@@ -70,13 +71,12 @@ class Terms extends AbstractQuery
     /**
      * Sets the minimum matching values.
      *
-     * @param int $minimum Minimum value
-     *
-     * @return $this
+     * @param int|string $minimum Minimum value
+     * @return  $this
      */
     public function setMinimumMatch($minimum)
     {
-        return $this->setParam('minimum_match', (int) $minimum);
+        return $this->setParam('minimum_match', $minimum);
     }
 
     /**

--- a/test/lib/Elastica/Test/Query/TermsTest.php
+++ b/test/lib/Elastica/Test/Query/TermsTest.php
@@ -36,14 +36,38 @@ class TermsTest extends BaseTest
         $this->assertEquals(3, $resultSet->count());
     }
 
+    public function provideMinimumArguments()
+    {
+        return [
+            [
+                3
+            ],
+            [
+                -2
+            ],
+            [
+                '75%'
+            ],
+            [
+                '-25%'
+            ],
+            [
+                '3<90%'
+            ],
+            [
+                '2<-25% 9<-3'
+            ]
+        ];
+    }
+
     /**
      * @group unit
+     * @dataProvider provideMinimumArguments
      */
-    public function testSetMinimum()
+    public function testSetMinimum($minimum)
     {
         $key = 'name';
         $terms = ['nicolas', 'ruflin'];
-        $minimum = 2;
 
         $query = new Terms($key, $terms);
         $query->setMinimumMatch($minimum);

